docs/WAIVERS.md

```diff
@@ -137,17 +108,87 @@ returning `False` if the RPM is not installed.
     rhel == 8 and 'some thing' in note and env('INFRA') == 'jenkins'
 ```
 
-### Expecting both `pass` and `fail`/`error`
+## Collecting a list of sections
 
-Sometimes, failures or errors are not reliable and happen only occassionally.
-This is why waivers are "permissive" by default - they only waive
-`fail`/`error`, but allow `pass` to pass through, only noting `waived pass`
-in the result note.
+Sections (as defined above) are gathered from multiple waiver files
+in alphanumerical order, and - within each file - from the top.
 
-To override this, pass `strict=True` to `Match()`, which tells the waiving code
-to trigger a waive failure when the waiver matches `pass`.
+This creates a one big unified sequential ordered list of sections
+that is later used for waiving.
 
-```python
-/some/result/name
-    Match(rhel >= 8, strict=True)
+## How a result is waived
+
+When `results.report()` is called by test code to report a new result,
+the waiving logic looks at the status (`pass`, `fail`, etc.).
```

Comment:
Don't mention test implementation details such as `results.report()`. Results reviewer might not be familiar with tests implementations and those lines basically say "see test implementation".

Use rather something like
```suggestion
For each individual result before reporting it, the waiving logic
looks at the status (`pass`, `fail`, etc.).
```
or
```suggestion
Before a result (e.g. rule result or overall test result) is reported,
the waiving logic looks at the status (`pass`, `fail`, etc.).
```